### ISSUE TITLE
`EncodeDriver` will only work on images if they have the same shape.

### ISSUE THREAD

**Issue by JoanFM** [jina-ai/jina#2203]
**Describe the problem**

`EncodeDriver` is using `all_content` property of a `DocumentSet` to extract the `content` of all the `docs` in the `set` in batches.

This has a hidden implication when working with `image` documents.

`all_contents` will try to `extract` the `blob` field from `Document` for all of them and stack them in batches

So from `B` images of shape (3, 224, 224), it will obtain an `np.array` of shape (B, 3, 224, 224).

This will only work if every `Document` has a `blob` of the same `shape`.

**Proposed solution**

For now, this should be clearly documented in the `documentation`. 

Fixing this issue and generalization could be too costly, because not -stacking the contents would break the `interface` of every `ImageEncoder` and detecting if every document has the same shape can have extra undesired effects

**Comment by Yongxuanzhang** [comment]
Different shape means, for example, A:(1, 224, 224) and B: (3, 224, 224)? As long as one dimension is different?

**Comment by JoanFM** [comment]
> Different shape means, for example, A:(1, 224, 224) and B: (3, 224, 224)? As long as one dimension is different?

yes, or (3, 224, 224) and (3, 125, 268) (which is a more common case for images (normally they all have 3 channels)

**Comment by Yongxuanzhang** [comment]
Do we encounter this before? Or we just use images with the same dimension? I thought we can craft the images to fit them into the same dimension values, as long as they have the same channels (1 or 3).
And will this be an issue for PDF search example, the images often have different sizes.

**Comment by JoanFM** [comment]
> Do we encounter this before? Or we just use images with the same dimension? I thought we can craft the images to fit them into the same dimension values, as long as they have the same channels (1 or 3).
> And will this be an issue for PDF search example, the images often have different sizes.

it was never encountered, but it should be documented. 

Exactly, so far we always use the ImageNormalizer

**Comment by nan-wang** [comment]
I think this is expected behavior, but I agree this should be documented.

**Comment by FionnD** [comment]
I am moving this issue to the doc repo
